blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineSearch.java

```diff
@@ -105,12 +106,15 @@ public String getType() {
                 }
             });
         }else{
+            GlobMatcher matcher = pipeline.contains("*") ? new GlobMatcher(pipeline) : null;
```

Comment:
A lot of code for `Pattern.compile(("\\Q"+pipeline.replace("*", "\\E.*\\Q")+"\\E").replace("\\Q\\E",""))` ;)